akka-typed-tests/src/test/java/akka/typed/javadsl/BehaviorBuilderTest.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.javadsl;
+
+import akka.typed.Behavior;
+import akka.typed.Terminated;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import static akka.typed.javadsl.Actor.same;
+import static akka.typed.javadsl.Actor.stopped;
+
+public class BehaviorBuilderTest extends JUnitSuite {
+    interface Message {
+    }
+
+    static class One implements Message {
+        public void foo() {}
+    }
+
+    @Test
+    public void shouldCompile() {
+        Behavior<Message> b = BehaviorBuilder.<Message>create()
+                .message(One.class, (ctx, o) -> {
+                    o.foo();
+                    return same();
+                })
+                .signal(Terminated.class, (ctx, t) -> {
+                    System.out.println("Terminating along with " + t.ref());
+                    return stopped();
+                })
+                .build();
```

Comment:
Would be API lovely to not need the `build` call, just using `List` though then won't do as we'll need to keep source code order of the matches (so traversing from the tail), we can work around that though I think